src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -660,21 +692,15 @@ else if (!"MQTT".equals(mqttConnect.protocolName().asString()) || mqttConnect.pr
                 {
                     reasonCode = UNSUPPORTED_PROTOCOL_VERSION;
                 }
-                else if (isSetWillFlag(flags))
-                {
-                    reasonCode = MALFORMED_PACKET;
-                }
                 else if (isSetBasicAuthentication(flags))
                 {
                     reasonCode = NOT_AUTHORIZED;
                 }
             }
 
-            if (reasonCode == 0)
+            if (reasonCode == SUCCESS)
```

Comment:
Let's put the error case first, with `break decode`, then success path follows below `if`.